lib/matplotlib/rcsetup.py

```diff
@@ -74,7 +74,8 @@ def f(s):
                 if allow_stringlist:
                     # Sometimes, a list of colors might be a single string
                     # of single-letter colornames. So give that a shot.
-                    return [scalar_validator(v.strip()) for v in s if v.strip()]
+                    return [scalar_validator(v.strip()) for v in s if
+                            v.strip()]
```

Comment:
The preferred way to split list comprehensions is before the keyword `for` (and/or `if` if additionally required:

~~~
return [scalar_validator(v.strip())
        for v in s if v.strip()]
~~~

This keeps the logical blocks together.